src/OrbitBase/ThreadPool.cpp

```diff
@@ -191,8 +191,8 @@ void ThreadPoolImpl::WorkerFunction() {
 
 };  // namespace
 
-std::unique_ptr<ThreadPool> ThreadPool::Create(size_t thread_pool_min_size,
+std::shared_ptr<ThreadPool> ThreadPool::Create(size_t thread_pool_min_size,
                                                size_t thread_pool_max_size,
                                                absl::Duration thread_ttl) {
-  return std::make_unique<ThreadPoolImpl>(thread_pool_min_size, thread_pool_max_size, thread_ttl);
+  return std::make_shared<ThreadPoolImpl>(thread_pool_min_size, thread_pool_max_size, thread_ttl);
```

Comment:
why did it become shared?

Reply:
This is explained in the commit message. `Executor` - the new base class - uses `std::enable_shared_from_this`.

Reply:
Maybe add a comment here as well, so that reader doesn't hunt for commit message. Is is far enough away from ScheduleAfter...

Reply:
Done. See #2411